Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1168,7 +1350,122 @@ namespace xr
                         inputSource.AimSpace.Pose.Orientation.W = location.pose.orientation.w;
                     }
                 }
-                
+
+                {
+                    // Initialize the gamepad object components
+                    auto& inputSource = InputSources[idx];
+                    auto controllerInfo = sessionImpl.ControllerInfo;
+                    if (inputSource.GamepadObject.buttons.size() != controllerInfo.CONTROLLER_BUTTONS)
+                    {
+                        inputSource.GamepadObject.buttons = std::vector<Button>(controllerInfo.CONTROLLER_BUTTONS);
```

Comment:
Just use the `.clear()` member function on `std::vector` rather than assigning an empty one. You'll avoid new allocations that way.